### ISSUE TITLE
[@types/ember bug] - destroy is not assignable

### ISSUE THREAD

**Issue by nightire** [typed-ember/ember-cli-typescript#310]
### Which package(s) does this problem pertain to?
  - [X] @types/ember

### What are instructions we can follow to reproduce the issue?
``` typescript
// you can replace this to Route / Controller / Service ... etc
import Component from '@ember/component';

export default class Foo extends Component {

  destroy() {
	// ...
  }

}
```

### Now about that bug. What did you expect to see?
It is expected to has no errors.

### What happened instead?
Typescript complains about:

```
Property 'destroy' in type 'HeaderSectionComponent' is not assignable to the same property in base type 'Component'.
Type '() => void' is not assignable to type '() => CoreObject'.
Type 'void' is not assignable to type 'CoreObject'.
```

**Comment by mike-north** [comment]
@nightire this is not a bug. According to the ember API docs for [`CoreObject#destroy`](https://www.emberjs.com/api/ember/release/classes/CoreObject/methods/destroy?anchor=destroy), the instance the object being destroyed is the return type

<a href="https://www.emberjs.com/api/ember/release/classes/CoreObject/methods/destroy?anchor=destroy">
<img width="745" alt="screen_shot_2018-09-15_at_11_08_27_am" src="https://user-images.githubusercontent.com/558005/45589284-d5d84500-b8d7-11e8-93d7-d9867780783d.png">
</a>

If you `return this;` from the destroy hook, it should type-check
```ts
// you can replace this to Route / Controller / Service ... etc
import Component from '@ember/component';

export default class Foo extends Component {

  destroy() {
	// ...
	return this; // the return type of `destroy()` is now `Foo`
  }

}
```
